### PR TITLE
Doxygen awesome

### PR DESCRIPTION
**Pull request by mhubii**
Adds modern style sheet [doxygen-awesome](https://github.com/jothepro/doxygen-awesome-css) to Doxygen documentation.

Note that this adds [doxygen-awesome](https://github.com/jothepro/doxygen-awesome-css) as a submodule into `doc`

Might need some help double checking if gh-pages compiled correctly

Old:
Screenshot from 2023-04-21 10-48-53

New, dark / bright mode depends on your browser settings:

Dark:
Screenshot from 2023-04-21 10-47-57

Bright:
Screenshot from 2023-04-21 10-55-33

### CODE REVIEW

**Comment by cmower** [comment]
Thanks @mhubii, looks great! 

I think you need to add the following to the `documentation.yaml` workflow.

```yaml
jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      # add this part VVVVVVVVVVVVVVVVVVVVVVV
      - name: Checkout repository
        uses: actions/checkout@v2
        with:
          submodules: 'recursive'
      # add this part ^^^^^^^^^^^^^^^^^^^^^^^^

      # Other steps in the workflow

```

This should be enough for the action to build `gh-pages`.

File: .github/workflows/documentation.yaml [hunk 1]
```diff
@@ -27,6 +27,8 @@ jobs:
     steps:
 
       - uses: actions/checkout@v2.3.4
+        with:
```
Comment (mhubii) [inline .github/workflows/documentation.yaml:30]: does this do?

Comment (cmower) [reply .github/workflows/documentation.yaml:30]: This should be fine, it maybe a good idea to upgrade to `actions/checkout@v3` (cf https://github.com/actions/checkout)

Comment (mhubii) [reply .github/workflows/documentation.yaml:30]: done

**Pull request merged**